app/settings/page.tsx

```diff
@@ -1,24 +1,74 @@
 "use client";
 
 import { useSession } from "next-auth/react";
-import { redirect } from "next/navigation";
+import { redirect, useRouter } from "next/navigation";
+import { updateAvaterForUser } from "@/lib/actions";
+import { getAvatarInitials } from "@/lib/utils";
 
 import { CenteredLayout } from "@/components/page-layouts";
-import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
 
 function SettingsPage() {
-  const session = useSession();
-  const userId = Number(session.data?.user?.id);
+  const router = useRouter();
+  const { data: session } = useSession();
+  if (!session?.user) redirect("/login");
 
-  if (!session.data?.user) redirect("/login");
+  const userId = Number(session?.user?.id);
+  if (Number.isNaN(userId)) return <p>Invalid user</p>;
 
-  if (Number.isNaN(userId)) {
-    console.log("User id is invalid: ", session);
-    return <p>Invalid user</p>;
-  }
+  const handleFileChange = async (event: any) => {
+    const file = event.target.files[0];
+    if (!file) return;
+    if (file.size > 250000) return; // TODO show error
+    const data = new FormData();
+    data.set("file", file);
+    await updateAvaterForUser(data);
+    router.refresh();
+  };
 
   return (
     <>
+      <Card>
+        <CardHeader>
+          <CardTitle>Profile picture</CardTitle>
+        </CardHeader>
+        <CardContent className="space-y-4">
+          <div className="col-span-6 ml-2 sm:col-span-4 md:mr-3">
+            <Input
+              type="file"
+              className="hidden"
+              id="photo"
+              onChange={handleFileChange}
+              accept="image/*"
+            />
+            <Label className="block text-gray-700 text-sm font-bold mb-2 text-center">
+              Profile Photo
+            </Label>
+
+            <div className="text-center">
+              <div className="mt-2">
+                <div className="flex justify-center items-center w-full h-full">
+                  <Avatar className="w-40 h-40 rounded-full bg-cover bg-no-repeat bg-center">
+                    {/* TODO setState(getCurrentUser().avatarUrl) */}
+                    <AvatarImage src={""} />
```

Comment:
can you get the user data and set the src here. Maybe get the userdata in the enclosing server component and pass it down as props. I've done similar things in the task-comments and tas-comments-client.tsx files